dbms/src/Common/SensitiveDataMasker.h

```diff
@@ -0,0 +1,173 @@
+
+
+#ifndef NDEBUG
+#    include <iostream>
+#endif
+
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+#include <re2/re2.h>
+#include <re2/stringpiece.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <Common/Exception.h>
+#include <Common/StringUtils/StringUtils.h>
+#include <common/logger_useful.h>
+
+
+//* TODO: option with hyperscan? https://software.intel.com/en-us/articles/why-and-how-to-replace-pcre-with-hyperscan
+// re2::set should also work quite fast, but it doesn't return the match position, only which regexp was matched
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPILE_REGEXP;
+    extern const int NO_ELEMENTS_IN_CONFIG;
+    extern const int INVALID_CONFIG_PARAMETER;
+
+}
+
+
+class SensitiveDataMasker
+{
+private:
+    class MaskingRule
+    {
+    private:
+        const std::string name;
+        const std::string replacement_string;
+        const std::string regexp_string;
+
+        const RE2 regexp;
+        const re2::StringPiece replacement;
+
+        std::atomic<std::uint64_t> matches_count = 0;
+
+        //options
+    public:
+        MaskingRule(const std::string & name, const std::string & regexpString, const std::string & replacementString)
+            : name(name)
+            , replacement_string(replacementString)
+            , regexp_string(regexpString)
+            , regexp(regexp_string, RE2::Quiet)
+            , replacement(replacement_string)
+        {
+            if (!regexp.ok())
+                throw DB::Exception(
+                    "SensitiveDataMasker: cannot compile re2: " + regexpString + ", error: " + regexp.error()
+                        + ". Look at https://github.com/google/re2/wiki/Syntax for reference.",
+                    DB::ErrorCodes::CANNOT_COMPILE_REGEXP);
+        }
+        int apply(std::string & data)
+        {
+            auto m = RE2::GlobalReplace(&data, regexp, replacement);
+            matches_count += m;
+            return m;
+        }
+
+        const std::string & getName() const { return name; }
+        const std::string & getReplacementString() const { return replacement_string; }
+        uint64_t getMatchesCount() const { return matches_count; }
+    };
+
+    std::vector<std::unique_ptr<MaskingRule>> all_masking_rules;
+
+public:
+    SensitiveDataMasker() {}
```

Comment:
Do we need this?

Reply:
Constructor w/o parameters is used in unit test. I've added XML based constructor and tests later (wasn't sure at the beginging uf poci API wilk allow me to do that easily) 

It's not needed for anything else, maybe i will just hide it in ifdef.